### PR TITLE
EOS-23221: Replace SSPL Logger with CORTX Logger Class

### PR DESCRIPTION
**Pull request by jjxsg**
## Problem Statement:

SSPL should use CORTX logger instead of implementing its own logger class.
## Solution Design:

Use CORTX logger and remove SSPL's own logging implementation.
## Coding:

* [x] Did you follow coding standard and verified it with flake8? 

* [x] Did your address all Codacy issues?

## Testing:

* [x] Unittest updated and all unittests are passing?

* [x] Sanity/Self tests are updated (if applicable)?

* [x] Manual testing done covering happy path and non-happy path?

## Integration:

* [ ] If there is any interface change, did you communicate it to other components?
* [ ] If there is any interface change, other component gate gatekeepers are ready to accept the change?
* [ ] Did you complete deployment test if applicable?

## PR checklist:
* [x] Did you add Jira number to the PR?

* [x] DCO and cla-signed?

### CODE REVIEW

File: low-level/framework/utils/service_logging.py [hunk 1]
```diff
@@ -19,130 +19,33 @@
...
 
-import logging.handlers
-import time
+import sys
 import os
+import syslog
+from cortx.utils.log import Log
+
+
+def init_logging(service_name, file_path, log_level="INFO"):
```
Comment (jjxsg) [inline low-level/framework/utils/service_logging.py:28]: > **cdeshmukh**: legacy DCS, there used to be another module along with SSPL daemon. we can safely rename that in new code, just keep service_name

Done

File: low-level/framework/utils/service_logging.py [hunk 2]
```diff
@@ -19,130 +19,33 @@
...
+    """Initialize logging for SSPL component."""
+    # Log rotation is configured within cortx-utils with
```
Comment (jjxsg) [inline low-level/framework/utils/service_logging.py:30]: > **cdeshmukh**: what about log rotation config? What are the defaults, if any exists, does it server SSPL needs?
> 
> **gargpalak048:** Log rotation is handled by cortx-utils itself.
> Default config in cortx-utils is,
> backup_count=10,
> file_size_in_mb=10,
> max_bytes = file_size_in_mb * 1024 * 1024
> most of the components like HA using this default config only, I think for SSPL also this will suffice
> 
> **cdeshmukh**: yes that suffice. but still add comments about default config around this code and also mention its recommended to refer to latest code for confirming config. that may help in future

Done

File: low-level/framework/utils/webservices.py [hunk 3]
```diff
@@ -47,6 +48,7 @@ class WebServices(object):
     def __init__(self):
         super(WebServices, self).__init__()
 
+        init_logging("sspl", SSPL_LOG_PATH)
```
Comment (jjxsg) [inline low-level/framework/utils/webservices.py:51]: > **cdeshmukh:** why separate init here again?
> 
> **gargpalak048:** We need to initialize logger obj in webservices.py file as well because realstor login api is called even before sspl_ll_d script.
>  
> **cdeshmukh:** How? sspl-ll_d has main() which starts the sspl daemon. You mean webservice class used before logging init? Then that init should simply happen first
> 

When [RealStorEnclosure](https://github.com/Seagate/cortx-monitor/blob/main/low-level/framework/platforms/realstor/realstor_enclosure.py) is imported, its `init` method is called in which we have a call to `login()` function which performs login through `realstor` API. So even before SSPL starts, we have a call to `realstor` API that uses functions from webservice class.

File: low-level/files/etc/rsyslog.d/0-iemfwd.conf [hunk 4]
```diff
@@ -1,30 +0,0 @@
-# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (sandeepsgt) [inline low-level/files/etc/rsyslog.d/0-iemfwd.conf:1]: This file is needed for journal to file IEM forwarding, IEM sensor will not work without this.
This file is not related to SSPL logs.

File: low-level/framework/utils/webservices.py (hunk 3 above)
Comment (cdeshmukh) [reply low-level/framework/utils/webservices.py:51]: hmm so in effect two logging handles created for same sspl log file !? We will have to fix this through separate patch, will not drag this one for that

File: low-level/files/etc/rsyslog.d/0-iemfwd.conf (hunk 4 above)
Comment (cdeshmukh) [reply low-level/files/etc/rsyslog.d/0-iemfwd.conf:1]: Ok, thanks for identifying. we should keep it then

File: low-level/files/etc/logrotate.d/iem_messages [hunk 5]
```diff
@@ -1,14 +0,0 @@
-/var/log/cortx/iem/iem_messages
```
Comment (cdeshmukh) [inline low-level/files/etc/logrotate.d/iem_messages:1]: as @sandeepsgt identified, we will need IEM logging (& logrotate) related config still, not related to SSPL logging

File: low-level/files/etc/rsyslog.d/0-iemfwd.conf (hunk 4 above)
Comment (jjxsg) [reply low-level/files/etc/rsyslog.d/0-iemfwd.conf]: Addressed, please check.

File: low-level/files/etc/logrotate.d/iem_messages (hunk 5 above)
Comment (jjxsg) [reply low-level/files/etc/logrotate.d/iem_messages]: Addressed, please check.

**Review by cdeshmukh — APPROVED** [review]

**Review by sumedhak27 — APPROVED** [review]

File: low-level/framework/utils/filestore.py [hunk 6]
```diff
@@ -117,15 +119,15 @@ def _load_json_file(self, key):
             except:
                 value = fh.read()
         except IOError as err:
-            logger.warning("I/O error[{0}] while loading data from file {1}): {2}"\
+            logger.warn("I/O error[{0}] while loading data from file {1}): {2}"\
```
Comment (satish-seagate) [inline low-level/framework/utils/filestore.py:122]: why we changed the warning to warn?
As I understand logging.warn has been deprecated since Python 3.3
https://stackoverflow.com/a/15655674

Comment (jjxsg) [reply low-level/framework/utils/filestore.py:122]: Yes, `warn` is depreciated in native python logger. But the logger object here is coming from a custom CORTX Log class which has `warn` method and not `warning`. 
Ref: https://github.com/Seagate/cortx-utils/blob/main/py-utils/src/utils/log.py

Comment (satish-seagate) [reply low-level/framework/utils/filestore.py:122]: Okay

**Review by satish-seagate — APPROVED** [review]

File: low-level/framework/utils/service_logging.py (hunk 1 above)
Comment (mariyappanp) [reply low-level/framework/utils/service_logging.py:28]: @jjxsg , While trying out with this change, logs are suppressed in sspl.log also same is writing into /var/log/messages.
